pkg/operator/resources/trafficsplitter/api.go

```diff
@@ -59,17 +59,45 @@ func UpdateAPI(apiConfig *userconfig.API, force bool) (*spec.API, string, error)
 			return nil, "", errors.Wrap(err, "upload api spec")
```

Comment:
(above) Should we remove from API Gateway if adding it failed, like we do in batch?